client/dist/static/js/app.js

```diff
@@ -0,0 +1,139 @@
+    set_hash({query: query});
+    event.stopPropagation();
+    event.preventDefault();
+});
+
+
+filterButton.click(function() {
+    event.stopPropagation();
+    event.preventDefault();
+    var current_query = process_query($megaSearchBar_query.val()).query;
+    var new_query_obj = get_filters();
+    new_query_obj.query = current_query;
+    var new_query = build_query(new_query_obj);
+    set_hash({query: new_query});
+});
+
+$(window).hashchange(function () {
+    var hash = get_hash();
+    $megaSearchBar_query.val(hash.query);
+    var query = process_query(hash.query);
+    set_filters(query);
+    if (!query.query) {
+        return;
+    }
+    sendQuery();
+}).hashchange();
+
+handleNextBtnClick = function () {
+    $results.goTo();
+};
+
+
+function sendQuery() {
+    var hash = get_hash();
+    var query = process_query(hash.query);
+    var page = parseInt(hash.page, 10) || 0;
+    var queryFrom = page * 10;
+
+    if (!query.query) {
+        $results.slideUp('fast');
+    } else {
+
+        var queryData = buildESQuery(query);
+        queryData.from = queryFrom;
+        
+        $.ajax('/search/search/_search', {
+            type: "POST",
+            data: JSON.stringify(queryData),
+            success: querySuccess,
+            dataType: 'json',
+            contentType: "application/json",
+        });
+    }
+}
+
+function querySuccess(data, status, xhr) {
+    var hash = get_hash();
+    var page = parseInt(hash.page, 10) || 0;
+
+    $('#results').html('');
+
+    var raw_results = data.hits.hits;
+    var results = $.map(raw_results, cleanResult);
+
+    var searchResults = {};
+    searchResults.hits = results;
+    if (data.hits.total >= (page + 1) * 10) {
+        hash.page = page + 1;
+        searchResults.nextUrl = make_hash(hash);
+    }
+    if (page > 0) {
+        hash.page = page - 1;
+        searchResults.prevUrl = make_hash(hash);
+    }
+
+    $("#results").html(template(searchResults));
+    $('.results_search-next').click(handleNextBtnClick);
+    $('#results').show();
+}
+
+function cleanResult(rawResult) {
+
+    var cleanedData = {
+        url: '',
+        repo: '',
+        title: '',
+        content: '',
+        index: '',
+        type: ''
+    };
+
+
+    var fields = rawResult.fields;
+    if (fields) {
+        $.extend(cleanedData, {
+            url: fields.url[0],
+            path: fields.path[0],
+            title: fields.title[0],
+            author: fields.author ? fields.author[0] : "",
+            assignee: fields.assignee ? fields.assignee[0] : "",
+            source: fields.source[0],
+            index: rawResult._index,
+            type: rawResult._type
+        });
+    }
+
+    var highlight = rawResult.highlight;
+    if (highlight && highlight.title) {
+        cleanedData.title = highlight.title[0];
+    }
+    if (highlight && highlight.content) {
+        cleanedData.content = highlight.content[0];
+    }
+
+    return cleanedData;
+
+}
```

Comment:
I'll leave the decision on checking in dist/ to y'all.

What this specifically does though is ensure that app.js also gets copied to dist/, so that sourcemap works and it's easy to debug the JS during development. This is a developer-time enhancement and has no effect on runtime. If there's a better way, I'm all ears. For me, the problem it solved was that during development, I was forced to work with the minified JS and that was not OK for debugging.